test/functional/feature_proxy.py

```diff
@@ -305,6 +308,32 @@ def networks_dict(d):
         assert_equal(n4['i2p']['reachable'], False)
         assert_equal(n4['cjdns']['reachable'], True)
 
+        self.stop_node(1)
+
+        self.log.info("Test passing invalid -proxy raises expected init error")
+        self.nodes[1].extra_args = ["-proxy=abc:def"]
+        msg = "Error: Invalid -proxy address or hostname: 'abc:def'"
+        self.nodes[1].assert_start_raises_init_error(expected_msg=msg)
+
+        self.log.info("Test passing invalid -onion raises expected init error")
+        self.nodes[1].extra_args = ["-onion=xyz:123"]
+        msg = "Error: Invalid -onion address or hostname: 'xyz:123'"
```

Comment:
`xyz:123` could be valid if `xyz` resolves. For example, `echo "127.0.0.1 xyz" >> /etc/hosts` bricks the test. Better use non-numeric port like in the `-proxy` test.

Reply:
Done

Reply:
(Good point!)